packages/hooking/hook.ts

```diff
@@ -14,33 +14,85 @@
  * limitations under the License.
  */
 
-/*eslint @typescript-eslint/no-explicit-any: 0 */
+// Stores package names and function names [packageName -> [fnName, ...], ...].
+// This structure is used to keep track of all functions seen during instrumentation and execution of the fuzzing run,
+// to determine which hooks have been applied, are available, and have not been applied.
+export class TrackedHooks {
+	hooks: Map<string, Set<string>>;
+	constructor() {
+		this.hooks = new Map();
```

Comment:
Field could directly be initialized to get rid of th constructor.

Reply:
done!